packages/language-server/src/__test__/artificial-panic.test.ts

```diff
@@ -0,0 +1,217 @@
+import { CodeActionParams, CompletionParams, DocumentFormattingParams } from 'vscode-languageserver'
+import { TextDocument, TextEdit } from 'vscode-languageserver-textdocument'
+import {
+  handleCodeActions,
+  handleCompletionRequest,
+  handleDiagnosticsRequest,
+  handleDocumentFormatting,
+} from '../MessageHandler'
+import { CURSOR_CHARACTER, findCursorPosition, getTextDocument } from './helper'
+
+import * as assert from 'assert'
+
+suite('Artificial Panics', () => {
+  const OLD_ENV = { ...process.env }
+
+  test('code actions', () => {
+    const fixturePath = './artificial-panic/schema.prisma'
+    const document: TextDocument = getTextDocument(fixturePath)
+
+    const params: CodeActionParams = {
+      textDocument: document,
+      range: {
+        start: { line: 0, character: 0 },
+        end: { line: 0, character: 0 },
+      },
+      context: {
+        diagnostics: [
+          {
+            range: {
+              start: { line: 0, character: 0 },
+              end: { line: 0, character: 0 },
+            },
+            message: 'dx msg',
+          },
+        ],
+      },
+    }
+
+    process.env.FORCE_PANIC_PRISMA_FMT = '1'
+
+    let calledCount = 0
+
+    const onError = () => {
+      calledCount += 1
+    }
```

Comment:
> track onError's argument, store it in the same outer scope as let calledCount, and assert its content in the catch block.

I remember that we did this initially but when looking back, I was confused as to why

Reply:
> add a comment explaining this is a hand-written jest.fn() mock equivalent

Is this necessary? It seems a little superfluous

Reply:
> Is this necessary?

I'm just thinking about future readers of that test

> I remember that we did this initially but when looking back, I was confused as to why

Because we are currently only testing that the `onError` function is called once, which isn't that useful.
We want to actually see how that function is called (i.e., the error message retrieved from the `wasm.debug_panic` is passed an argument to that `onError` function, and we should test that)

Reply:
Sorted!